pkg/specgenutil/specgenutil_test.go

```diff
@@ -154,3 +156,64 @@ func TestParseLinuxResourcesDeviceAccess(t *testing.T) {
 	_, err = parseLinuxResourcesDeviceAccess("a *:-3 r")
 	assert.NotNil(t, err, "err is not nil")
 }
+
+func TestGenRlimits(t *testing.T) {
+	testLimits := map[string]string{
+		"core":       "1:2",
+		"cpu":        "1:2",
+		"data":       "1:2",
+		"fsize":      "1:2",
+		"locks":      "1:2",
+		"memlock":    "1:2",
+		"msgqueue":   "1:2",
+		"nice":       "1:2",
+		"nofile":     "1:2",
+		"nproc":      "1:2",
+		"rss":        "1:2",
+		"rtprio":     "1:2",
+		"rttime":     "1:2",
+		"sigpending": "1:2",
+		"stack":      "1:2",
+	}
+
+	lowerCaseLimits := make([]string, 0, len(testLimits))
+	upperCaseLimitsWithPrefix := make([]string, 0, len(testLimits))
+	for name, limit := range testLimits {
+		lowerCaseLimits = append(lowerCaseLimits, fmt.Sprintf("%s=%s", name, limit))
+		upperCaseLimitsWithPrefix = append(upperCaseLimitsWithPrefix, strings.ToUpper(fmt.Sprintf("%s%s=%s", rlimitPrefix, name, limit)))
+	}
+
+	parsedLimits, err := GenRlimits(lowerCaseLimits)
+	assert.Nil(t, err, "err is nil")
+
+	for _, limit := range parsedLimits {
+		val, ok := testLimits[limit.Type]
+		assert.True(t, ok, "%s matched", limit.Type)
+		assert.Equal(t, val, fmt.Sprintf("%d:%d", limit.Soft, limit.Hard), "soft and hard limits are equal")
+	}
+
+	parsedLimits, err = GenRlimits(upperCaseLimitsWithPrefix)
+	assert.Nil(t, err, "err is nil")
+
+	for _, limit := range parsedLimits {
+		val, ok := testLimits[limit.Type]
+		assert.True(t, ok, "%s matched", limit.Type)
+		assert.Equal(t, val, fmt.Sprintf("%d:%d", limit.Soft, limit.Hard), "soft and hard limits are equal")
+	}
+
+	// go-utils module has no tests for the parser.
+	_, err = GenRlimits([]string{"foo=1:1"})
+	assert.NotNil(t, err, "err is not nil")
+
+	_, err = GenRlimits([]string{"RLIMIT_FOO=1:1"})
+	assert.NotNil(t, err, "err is not nil")
+
+	_, err = GenRlimits([]string{"nofile"})
+	assert.NotNil(t, err, "err is not nil")
+
+	_, err = GenRlimits([]string{"nofile=bar"})
+	assert.NotNil(t, err, "err is not nil")
+
+	_, err = GenRlimits([]string{"nofile=bar:buzz"})
+	assert.NotNil(t, err, "err is not nil")
```

Comment:
consider using assert.Error() for these, this will provide a much friendlier error message in case of an failure